### PR TITLE
added lv_dialog

### PR DESCRIPTION
**Pull request by mariotaku**
### Description of the feature or fix

Added general purpose dialog as mentioned in #2560.

A `msgbox` like widget, but you can put any content into it.

image

### Checkpoints
- [x] Follow the [styling guide](https://github.com/lvgl/lvgl/blob/master/docs/CODING_STYLE.md)
- [ ] Update [CHANGELOG.md](https://github.com/lvgl/lvgl/blob/master/docs/CHANGELOG.md)
- [ ] Update the documentation

### CODE REVIEW

**Comment by kisvegabor** [comment]
Unless I miss something it's the same as `lv_msgbox` except adding a "content box", right?
If so, IMO we should rather improve the message box with it. It shouldn't break anything.

File: src/extra/widgets/msgbox/lv_msgbox.c [hunk 1]
```diff
@@ -87,10 +103,15 @@ lv_obj_t * lv_msgbox_create(lv_obj_t * parent, const char * title, const char *
...
+    bool has_txt = txt && strlen(txt) > 0;
+    if (has_txt) {
+        mbox->text = lv_label_create(mbox->content);
+        lv_label_set_text(mbox->text, txt);
+        lv_label_set_long_mode(mbox->text, LV_LABEL_LONG_WRAP);
+        lv_obj_set_width(mbox->text, lv_pct(100));
+    }
 
     if(btn_txts) {
         mbox->btns = lv_btnmatrix_create(obj);
```
Comment (kisvegabor) [inline src/extra/widgets/msgbox/lv_msgbox.c:117]: Shouldn't it be created on `content` too?

Comment (mariotaku) [reply src/extra/widgets/msgbox/lv_msgbox.c:117]: Text is now placed within `mbox->content`. What do you think?

Comment (kisvegabor) [reply src/extra/widgets/msgbox/lv_msgbox.c:117]: I mean the btnmatrix.

Comment (mariotaku) [reply src/extra/widgets/msgbox/lv_msgbox.c:117]: @kisvegabor IMO it's better to put buttons outside. If the user want to, they can add complex layout inside `content` while still having consistent buttons. Also this is how Android design its dialog.

File: src/extra/themes/default/lv_theme_default.c [hunk 2]
```diff
@@ -955,8 +955,15 @@ static void theme_apply(lv_theme_t * th, lv_obj_t * obj)
     else if(lv_obj_check_type(obj, &lv_msgbox_class)) {
         lv_obj_add_style(obj, &styles->card, 0);
         lv_obj_add_style(obj, &styles->msgbox_bg, 0);
+        lv_obj_set_style_flex_main_place(obj, LV_FLEX_ALIGN_START, 0);
```
Comment (kisvegabor) [inline src/extra/themes/default/lv_theme_default.c]: In the theme, I'd prefer adding only style. Local style needs more RAM and can't be overwritten by "normal" styles later. 

However doesn't it work to use  the default `LV_FLEX_ALIGN_START` on each align?

File: src/extra/themes/default/lv_theme_default.c [hunk 3]
```diff
@@ -955,8 +955,15 @@ static void theme_apply(lv_theme_t * th, lv_obj_t * obj)
...
+        lv_obj_set_style_flex_cross_place(obj, LV_FLEX_ALIGN_CENTER, 0);
+        lv_obj_set_style_flex_track_place(obj, LV_FLEX_ALIGN_START, 0);
         return;
     }
+    else if(lv_obj_check_type(obj, &lv_msgbox_backdrop_class)) {
+        lv_obj_set_style_bg_color(obj, lv_palette_main(LV_PALETTE_GREY), 0);
```
Comment (kisvegabor) [inline src/extra/themes/default/lv_theme_default.c]: Here `lv_obj_add_style` would be really better.

File: src/extra/widgets/msgbox/lv_msgbox.c (hunk 1 above)
Comment (kisvegabor) [reply src/extra/widgets/msgbox/lv_msgbox.c:117]: Makes sense.

File: src/extra/themes/default/lv_theme_default.c (hunk 2 above)
Comment (mariotaku) [reply src/extra/themes/default/lv_theme_default.c]: I see. Default style should work. Let me try that one.

**Comment by kisvegabor** [comment]
Looks good, thank you!

**Pull request merged**